include/deal.II/base/array_view.h

```diff
@@ -157,6 +157,17 @@ class ArrayView
    */
   ArrayView (std::vector<typename std::remove_cv<value_type>::type> &vector);
 
+  /**
+   * Compare two ArrayView objects of the same type. Two objects are considered
+   * equal if they have the same size and the same starting pointer.
+   */
+  bool operator == (const ArrayView<const value_type> &other_view) const;
```

Comment:
Shouldn't the argument be of type `const ArrayView<value_type> &`? What happens if you had
```
  ArrayView<int> a;
  a == a;
```
? This operator could not immediately be called, but it is conceivable that the second argument would be converted to an `ArrayView<const int>` that has the same pointer and length as `a`, of course.

Reply:
Using `const ArrayView<value_type> &` I can't compare `ArrayView<T>` with `ArrayView<const T>` because there is no implicit conversion operator from the firsts type to the second.
Comparing  `ArrayView<const T>` with `ArrayView<T>` would also work as `ArrayView<T>` is convertible to `ArrayView<const T>`.

That is the reason for using `const ArrayView<const value_type> &` instead; we can always convert to an `ArrayView` with `const` `value_type`.

If you prefer, I can also declare a comparison using  `ArrayView<T>` avoiding the implicit conversion.

Reply:
I think you probably want to have two `operator==`, one to compare with the `const` and one to compare with the non-`const` version. These are, respectively, `ArrayView<const value_type>` and `ArrayView<typename std::remove_cv<value_type>::type>`. 

Since one of them is guaranteed to be the current type, the comparison will never require a copy/conversion of the object.